src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDynamicInterfaceCastableImplementation.Fixer.cs

```diff
@@ -0,0 +1,203 @@
+                {
+                    propertyDecl = generator.WithSetAccessorStatements(propertyDecl, defaultMethodBodyStatements);
+                }
+
+                return propertyDecl;
+            }
+        }
+
+        private static SyntaxNode GenerateEventImplementation(
+            IEventSymbol evt,
+            SyntaxNode declaration,
+            SyntaxGenerator generator,
+            SyntaxNode[] defaultMethodBodyStatements)
+        {
+            SyntaxNode eventDecl = generator.CustomEventDeclaration(evt);
+            eventDecl = generator.WithModifiers(eventDecl, generator.GetModifiers(declaration).WithIsAbstract(false));
+
+            // Explicitly use the C# syntax APIs to work around https://github.com/dotnet/roslyn/issues/53649
+            EventDeclarationSyntax eventDeclaration = (EventDeclarationSyntax)eventDecl;
+
+            return eventDeclaration.WithAccessorList(
+                SyntaxFactory.AccessorList(
+                    SyntaxFactory.List(
+                new[]
+                {
+                        generator.WithStatements(generator.GetAccessor(eventDecl, DeclarationKind.AddAccessor), defaultMethodBodyStatements),
+                        generator.WithStatements(generator.GetAccessor(eventDecl, DeclarationKind.RemoveAccessor), defaultMethodBodyStatements),
+                })));
+        }
+
+        protected override async Task<Document> SealMemberDeclaredOnImplementationType(
+    SyntaxNode declaration,
+    Document document,
+    CancellationToken ct)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, ct).ConfigureAwait(false);
+            var generator = editor.Generator;
+            var defaultMethodBodyStatements = generator.DefaultMethodBody(editor.SemanticModel.Compilation).ToArray();
+
+            ISymbol symbol = editor.SemanticModel.GetDeclaredSymbol(declaration, ct);
+            if (declaration.IsKind(SyntaxKind.EventFieldDeclaration))
+            {
+                // We'll only end up here with one event in the event field declaration syntax.
+                EventFieldDeclarationSyntax eventField = (EventFieldDeclarationSyntax)declaration;
+                symbol = editor.SemanticModel.GetDeclaredSymbol(eventField.Declaration.Variables[0], ct);
+            }
+
+            editor.ReplaceNode(generator.GetDeclaration(declaration), SealMemberDeclaration(declaration, symbol));
+            return editor.GetChangedDocument();
+
+            SyntaxNode SealMemberDeclaration(SyntaxNode declaration, ISymbol symbol)
+            {
+                switch (symbol.Kind)
+                {
+                    case SymbolKind.Method:
+                        {
+                            DeclarationModifiers modifiers = generator.GetModifiers(declaration)
+                               .WithIsAbstract(false)
+                               .WithIsVirtual(false)
+                               .WithIsSealed(true);
```

Comment:
It looks like all the 3 switch cases are using this. Could it be moved outside the switch?